tests/test_functions.py

```diff
@@ -0,0 +1,197 @@
+import sys
+import os
+import unittest
+import tempfile
+
+from csvjoin import functions
+from io import StringIO
+
+
+class Capturing(list):
+    """
+    Captures the output from the print() function.
+
+    Only for tests.
+    """
+    def __enter__(self):
+        self._stdout = sys.stdout
+        sys.stdout = self._stringio = StringIO()
+        return self
+
+    def __exit__(self, *args):
+        self.extend(self._stringio.getvalue().splitlines())
+        del self._stringio  # free up some memory
+        sys.stdout = self._stdout
```

Comment:
This is really weird, I don't understand what this class is doing at all. Also, why do you need to manually free memory in debugging?